src/main/java/com/google/cloud/resourcemanager/ConstraintInfo.java

```diff
@@ -0,0 +1,315 @@
+      this.description = description;
+      return this;
+    }
+
+    @Override
+    public Builder setDisplayName(String displayName) {
+      this.displayName = displayName;
+      return this;
+    }
+
+    @Override
+    public Builder setListConstraint(ListConstraint listConstraint) {
+      this.listConstraint = listConstraint;
+      return this;
+    }
+
+    @Override
+    public Builder setName(String name) {
+      this.name = name;
+      return this;
+    }
+
+    @Override
+    public Builder setVersion(Integer version) {
+      this.version = version;
+      return this;
+    }
+
+    @Override
+    public ConstraintInfo build() {
+      return new ConstraintInfo(this);
+    }
+  }
+
+  ConstraintInfo(BuilderImpl builder) {
+    this.booleanConstraint = builder.booleanConstraint;
+    this.constraintDefault = builder.constraintDefault;
+    this.description = builder.description;
+    this.displayName = builder.displayName;
+    this.listConstraint = builder.listConstraint;
+    this.name = builder.name;
+    this.version = builder.version;
+  }
+
+  public BooleanConstraint getBooleanConstraint() {
+    return booleanConstraint;
+  }
+
+  public String getConstraintDefault() {
+    return constraintDefault;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+
+  public String getDisplayName() {
+    return displayName;
+  }
+
+  public ListConstraint getListConstraint() {
+    return listConstraint;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public Integer getVersion() {
+    return version;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    return obj == this
+        || obj != null
+            && obj.getClass().equals(ConstraintInfo.class)
+            && Objects.equals(toPb(), ((ConstraintInfo) obj).toPb());
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+        booleanConstraint,
+        constraintDefault,
+        description,
+        displayName,
+        listConstraint,
+        name,
+        version);
+  }
+
+  public static Builder newBuilder(String name) {
+    return new BuilderImpl(name);
+  }
+
+  public Builder toBuilder() {
+    return new BuilderImpl(this);
+  }
+
+  com.google.api.services.cloudresourcemanager.model.Constraint toPb() {
+    com.google.api.services.cloudresourcemanager.model.Constraint constraintPb =
+        new com.google.api.services.cloudresourcemanager.model.Constraint();
+    constraintPb.setBooleanConstraint(booleanConstraint);
+    constraintPb.setConstraintDefault(constraintDefault);
+    constraintPb.setDescription(description);
+    constraintPb.setDisplayName(displayName);
+    if (listConstraint != null) {
+      constraintPb.setListConstraint(listConstraint.toPb());
+    }
+    constraintPb.setName(name);
+    constraintPb.setVersion(version);
+    return constraintPb;
+  }
+
+  static ConstraintInfo fromPb(
+      com.google.api.services.cloudresourcemanager.model.Constraint constraintPb) {
```

Comment:
Please try to avoid reusing class names. It's confusing and hard to follow.

Reply:
done